lib/screens/v2/candidate_generator/widgets/cr_departures.ex

```diff
@@ -0,0 +1,60 @@
+defmodule Screens.V2.CandidateGenerator.Widgets.CRDepartures do
+  @moduledoc false
+
+  alias Screens.Config.Screen
+  alias Screens.Config.V2.{CRDepartures, PreFare}
+  alias Screens.Stops.Stop
+  alias Screens.V2.Departure
+  alias Screens.V2.WidgetInstance.CRDepartures, as: CRDeparturesWidget
+  alias Screens.V2.WidgetInstance.{DeparturesNoData, OvernightDepartures}
+
+  def departures_instances(
+        config,
+        fetch_departures_fn \\ &fetch_departures/2,
+        fetch_stop_name_fn \\ &Stop.fetch_stop_name/1
+      )
+
+  def departures_instances(
+        %Screen{app_params: %app{cr_departures: %CRDepartures{} = cr_departures}} = config,
+        fetch_departures_fn,
+        fetch_stop_name_fn
+      )
+      when app in [PreFare] do
+    opts = %{include_schedules: true}
+
+    params = %{
+      direction_id: cr_departures.direction_to_destination,
+      route_type: :rail,
+      stop_ids: [cr_departures.station]
+    }
+
+    destination = fetch_stop_name_fn.(cr_departures.destination)
+
+    {:ok, departures_data} = fetch_departures_fn.(opts, params)
+
+    # The Overnight and NoData widgets may not be relevant here
```

Comment:
Oops, I just saw this code comment after writing my overly-detailed PR comment below. Feel free to disregard whatever I just said, or use whatever part of it may be useful.